src/osd/OSD.cc

```diff
@@ -9488,3 +9505,32 @@ void OSD::PeeringWQ::_dequeue(list<PG*> *out) {
   }
   in_use.insert(out->begin(), out->end());
 }
+
+namespace ceph { 
+namespace osd_cmds { 
+namespace admin {
+
+int heap(CephContext& cct, cmdmap_t& cmdmap, Formatter& f, std::ostream& os)
+{
+  if (!ceph_using_tcmalloc()) {
+        os << "could not issue heap profiler command -- not using tcmalloc!";
+        return -EOPNOTSUPP;
+  }
+  
+  string cmd;
+  std::vector<std::string> cmd_vec;
+ 
+  if (false == cmd_getval(&cct, cmdmap, "heapcmd", cmd)) {
```

Comment:
usually we don't do yoda conditions - any reason to avoid !cmd_getval(...) here?